src/realm/timestamp.hpp

```diff
@@ -75,6 +76,15 @@ class Timestamp {
         : m_is_null(true)
     {
     }
+    template <class T>
+    Timestamp(std::chrono::time_point<T> t)
+        : m_is_null(false)
+    {
+        auto time_since_epoc = t.time_since_epoch();
```

Comment:
Core should not interact with the system clock in any way; that should be done by the binding which is "closer" to the application and thus "closer" to whatever time standard/system the application uses, etc, etc.